### PR TITLE
Feat: API to retrieve sessions belonging to a user

### PR DESCRIPTION
**Pull request by Bahugunajii**
### Description
- Now we have a support of an API which will allow us to fetch all the live classes and quiz sessions belonging to the logged in user in gurukul

### Checklist
- [x] Self-testing

### CODE REVIEW

File: lib/dbservice/groups.ex [hunk 1]
```diff
@@ -31,6 +31,32 @@ defmodule Dbservice.Groups do
...
+  Gets group by group_id.
+  Raises `Ecto.NoResultsError` if the Group does not exist.
+  ## Examples
+      iex> get_group_by_group_id(123)
+      %Group{}
+      iex> get_group_by_group_id(456)
+      ** (Ecto.NoResultsError)
+  """
+  def get_group_by_group_id(group_id) do
+    Repo.get_by(Group, id: group_id, type: "batch")
```
Comment (suryabulusu) [inline lib/dbservice/groups.ex:44]: why hardcode `batch` here? maybe this can be in input too?

File: lib/dbservice/groups.ex [hunk 2]
```diff
@@ -31,6 +31,32 @@ defmodule Dbservice.Groups do
...
+  Gets group by child_id.
+  Raises `Ecto.NoResultsError` if the Group does not exist.
+  ## Examples
+      iex> get_group_by_child_id(123)
+      %Group{}
+      iex> get_group_by_child_id(456)
+      ** (Ecto.NoResultsError)
+  """
+  def get_group_by_child_id(child_id) do
+    Repo.get_by(Group, child_id: child_id, type: "batch")
```
Comment (suryabulusu) [inline lib/dbservice/groups.ex:57]: same comment as above

File: lib/dbservice_web/controllers/user_controller.ex [hunk 3]
```diff
@@ -170,4 +176,45 @@ defmodule DbserviceWeb.UserController do
...
+
+    case Groups.get_group_by_group_id(group_id) do
+      nil -> []
+      group -> get_group(group, quiz_flag)
+    end
+  end
+
+  defp get_group(group, quiz_flag) do
+    child_id = group.child_id
+    batch = Batches.get_batch!(child_id)
```
Comment (suryabulusu) [inline lib/dbservice_web/controllers/user_controller.ex:203]: here we assume that type is always batch.. maybe its fine for now but later on we will have to expand to other types

**Review by suryabulusu — COMMENTED** [review]
left comments

File: lib/dbservice/groups.ex (hunk 1 above)
Comment (Bahugunajii) [reply lib/dbservice/groups.ex:44]: Currently we always have sessions for a batch. Later if there is a need where session are associated with a program, product, etc. then we can have a more generic function where type can be just passed as an argument of this function.

Comment (suryabulusu) [reply lib/dbservice/groups.ex:44]: got it

Comment (suryabulusu) [reply lib/dbservice/groups.ex:44]: i'm making an issue so that we can remember this for future

**Review by suryabulusu — APPROVED** [review]
looks good. made an issue on repo for future reference

**Pull request merged**